src/main/java/core/basesyntax/OperationStrategyImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
+    }
+}

```

Comment:
The `OperationStrategyImpl` class should be placed in the `strategy` package according to the checklist.